### PR TITLE
Fix bug where default playlist were disappearing when changing username

### PR DESCRIPTION
**Pull request by BartLucien**
If merged, this PR will remove the bug that would remove all playlist from a non google user when changing its username.

### CODE REVIEW

File: app/src/main/java/com/github/fribourgsdp/radio/data/view/UserProfileActivity.kt [hunk 1]
```diff
@@ -122,9 +122,11 @@ open class UserProfileActivity : MyAppCompatActivity(), KeepOrDismissPlaylistDia
         if(user.isGoogleUser){
             db.setUser(user.id,user)
         }else{
+            val userPlaylists = user.getPlaylists()
             val userWithoutPlaylist = user
-            userWithoutPlaylist.removePlaylists(user.getPlaylists())
+            userWithoutPlaylist.removePlaylists(userPlaylists)
             db.setUser(user.id,userWithoutPlaylist)
+            user.addPlaylists(userPlaylists)
```
Comment (VictorSchneuwly) [inline app/src/main/java/com/github/fribourgsdp/radio/data/view/UserProfileActivity.kt:129]: Why remove the playlist to then add them later?

Comment (BartLucien) [reply app/src/main/java/com/github/fribourgsdp/radio/data/view/UserProfileActivity.kt:129]: When an user is not signed in, only his username is saved online, not his playlists, this is why I need to remove all the  playlists from the user before saving it. I have to reput the playlist in the original user as in reality userWithoutPlaylist and  user refer to the same entity. I create the UserWithoutPlaylist just for clarity reason.

Comment (Flamantrose321) [inline app/src/main/java/com/github/fribourgsdp/radio/data/view/UserProfileActivity.kt:129]: Why does it save the name in the database even if you are not connected to Google?

Comment (BartLucien) [reply app/src/main/java/com/github/fribourgsdp/radio/data/view/UserProfileActivity.kt:129]: The idea was to save the name of the user on the server. After discussion with @VictorSchneuwly, it seems that we don't use it at all, as a user gives its name directly to the lobby. This behavior and uniformisation of what a offline user can do will be treated in my other task, where I have to add the offline/online behavior.

**Review by VictorSchneuwly — APPROVED** [review]
Seems good to me now

**Review by Flamantrose321 — APPROVED** [review]

**Pull request merged**